USAGE.md

```diff
@@ -3,34 +3,80 @@
 ## 1. Vault Enterprise License
 Retrieve a [Vault Enterprise trial license](https://www.hashicorp.com/products/vault/trial).
 
-Export the license in your shell environment:
-```yaml
-export VAULT_LICENSE=02MV4UU..
-```
-
-## 2. Start Containers
+## 2. Setup environment
+Export environment variables in a shell for
+
+* the Vault license
+  ```vash
+  export VAULT_LICENSE=02MV4UU..
+  ```
+* the runtime you're going to use
+  ```bash
+  export CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker}
+  ```
+  *The default is `docker`. If podman is going to be used, change this to `podman`.*
+* the internal domain containers can be addressed with
+  ```bash
+  if [[ -z ${CONTAINER_RUNTIME##docker} ]];then export CONTAINER_DOMAIN=docker.internal; else export CONTAINER_DOMAIN=dns.podman; fi
+  ```
+  *uses `docker.internal` if docker is going to be used and `dns.podman` if podman. This does not need modification unless another domain name is required.*
+* the realm domain for the freeipa server
+  ```bash
+  export IPA_REALM=${CONTAINER_DOMAIN^^}
+  ```
+
+Or create a `.env` file for `(docker|podman)-compose`
+```bash
+# Example for podman
+CONTAINER_RUNTIME=podman
+VAULT_LICENSE=02MV4UU..
+echo CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker} > .env
+if [[ -z ${CONTAINER_RUNTIME##docker} ]];then echo CONTAINER_DOMAIN=docker.internal; else echo CONTAINER_DOMAIN=dns.podman; fi >> .env
+echo VAULT_LICENSE=${VAULT_LICENSE} >> .env
+echo IPA_REALM=${CONTAINER_DOMAIN^^} >> .env
+```
+
+## 3. Start Containers
 
 Start FreeIPA, Keycloak, Vault and K3s:
 ```bash
-docker-compose up -d
+sudo ${CONTAINER_RUNTIME}-compose up -d
 ```
 
-## 3. Setup Resolution of Domain Names
+## 4. Setup Resolution of Domain Names
+
+You can either add a static record for every container to your `/etc/hosts` file or configure systemd-resolved service with `resolvectl` to use the container environment internal DNS (recommended).
+
+### Option 1: Configuration using container internal dns (recommended)
+First, find out which interface is used
+```bash
+CONTAINER_RUNTIME_IF=$(sudo ${CONTAINER_RUNTIME} network inspect vault-playground_vault-playground | jq -j '.[].network_interface')
+```
+which IP it has assigned
+```bash
+CONTAINER_RUNTIME_IF_IP=$(ip -j -p -br -4 addr show dev ${CONTAINER_RUNTIME_IF} | jq -j '.[].addr_info | .[].local')
+```
+and then configure this interface to resolve the domain specified
+```bash
+sudo -E resolvectl dns ${CONTAINER_RUNTIME_IF} ${CONTAINER_RUNTIME_IF_IP}
```

Comment:
Work with OpenSuse Tumbleweed and NetworkManager is the default (https://en.opensuse.org/Systemd-networkd). I tried to follow the instructions to switch to `systemd-networkd`/`systemd-resolved`. Managed to brick my network setup. Was glad that I could do `snapper rollback`. Will probably just continue using the (cheap) `/etc/hosts` approach below :see_no_evil: But very nice that we have some more Fedore/Systemd specific instructions here :muscle:

Reply:
okay, we could also try to set search domain and dns in network manager. But I believe to recall that this didn't work as expected. something like
```
nmcli c mod ${CONTAINER_RUNTIME_IF} ipv4.dns ${CONTAINER_RUNTIME_IF_IP}
nmcli c mod ${CONTAINER_RUNTIME_IF} ipv4.dns-search ${CONTAINER_DOMAIN}
```

Reply:
That sounds like a reasonable idea. Need more time to actually test this

Reply:
Thanks for your suggestion @smirta. I added this in https://github.com/adfinis/vault-playground/pull/9/commits/47e94a3a10eccd62f9150671ca8abcee6b13cf86.